jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/IntervalGraphRecognizer.java

```diff
@@ -242,13 +248,32 @@ public boolean isIntervalGraph()
     }
 
     /**
-     * Returns an interval graph representation of the graph.
+     * Returns the list of all intervals sorted by starting point, or null, if the graph was not an interval graph.
+     *
+     * @return The list of all intervals sorted by starting point, or null, if the graph was not an interval graph.
+     */
+    public List<Interval<Integer>> getIntervalsSortedByStartingPoint()
```

Comment:
As it is very important for efficiency reasons that this is an ArrayList, this should have return value ArrayList instead of List.